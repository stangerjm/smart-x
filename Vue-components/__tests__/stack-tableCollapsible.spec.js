import {mount} from 'vue-test-utils'
import stackTableCollapsible from '../components/stack-tableCollapsible'
import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import Vue from "vue";

let tableData = [
  {
    'id': 1,
    'code': '01',
    'name': 'test',
    'testProp': 'test',
    'anotherProp': 'more',
    'detailsContext': 'Manufacturer',
    'deleteContext': 'County'
  },
  {
    'id': 2,
    'code': '02',
    'name': 'SW Washington',
    'testProp': 'test',
    'anotherProp': 'more',
    'editContext': 'Bottle',
    'deleteContext': 'Google'
  },
  {
    'id': 3,
    'code': '03',
    'name': 'Central Washington',
    'testProp': 'test',
    'anotherProp': 'more',
    'editContext': 'Else',
    'detailsContext': 'Test'
  }
];

Vue.mixin({
  methods: {
    formatFromCamelCase: text => {
      return text
        .replace(/([A-Z])/g, ' $1')
        .replace(/^./, (substr) => {
          return substr.toUpperCase();
        });
    }
  }
});

function mountTableCollapsible(props) {
  if (!props) {
    props = {
      propsData: {
        tableData: tableData,
        defaultContext: 'Region',
        title: 'Test'
      }
    }
  }

  return mount(stackTableCollapsible, props);
}

describe('stack-tableCollapsible.vue', () => {
  it('has "stack-tableCollapsible" as a class value', () => {
    const wrapper = mountTableCollapsible();

    expect(wrapper.vm.$el.className).toEqual('stack-tableCollapsible');
  });

  it('has the same HTML structure', () => {
    const renderer = createRenderer();
    const wrapper = shallow(stackTableCollapsible, {
      propsData: {
        tableData: tableData,
        defaultContext: 'Region',
        title: 'Test'
      }
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err);
      }
      expect(str).toMatchSnapshot();
    });
  });
});