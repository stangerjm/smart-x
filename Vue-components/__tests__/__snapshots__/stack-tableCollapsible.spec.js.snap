// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stack-tableCollapsible.vue has the same HTML structure 1`] = `
<div class="stack-tableCollapsible">
  <!---->
</div>
`;
